### PR TITLE
Fix zero-length repeat delegates ignoring initial delay

### PR DESCRIPTION
**Pull request by peppy**
A bit of an oversight that we somehow hadn't encountered until now.

The conditional was far too early in the flow, meaning that `ExecutionTime` is completely ignored in the case that `RepeatInterval` is zero.

### CODE REVIEW

File: osu.Framework/Threading/Scheduler.cs [hunk 1]
```diff
@@ -133,7 +133,9 @@ private void queueTimedTasks()
                             tasksToSchedule.Add(sd);
                         }
 
-                        if (!sd.Completed)
+                        if (sd.RepeatInterval == 0)
+                            perUpdateTasks.Add(sd);
```
Comment (smoogipoo) [inline osu.Framework/Threading/Scheduler.cs:137]: Isn't this going to be scheduled twice, because of the above block?

Comment (smoogipoo) [reply osu.Framework/Threading/Scheduler.cs:137]: Also, does this actually work the first time around? Shouldn't the `runQueue.Enqueue()` still happen?

Comment (peppy) [reply osu.Framework/Threading/Scheduler.cs:137]: The first run is handled by `Update`
image

I'll check / add test coverage of the double case, if that's actually a thing.

Comment (peppy) [reply osu.Framework/Threading/Scheduler.cs:137]: Regarding your first comment, I don't believe so? That code is adding the next iteration of the repeat.

Comment (smoogipoo) [reply osu.Framework/Threading/Scheduler.cs:137]: image

This is what I'm referring to.

Comment (peppy) [reply osu.Framework/Threading/Scheduler.cs:137]: aha, i see.

**Review by smoogipoo — APPROVED** [review]
lgtm

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**